src/parser.rs

```diff
@@ -2302,6 +2323,166 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub fn parse_create_trigger(&mut self, or_replace: bool) -> Result<Statement, ParserError> {
+        if dialect_of!(self is PostgreSqlDialect) {
+            let name = self.parse_object_name()?;
+            let period = self.parse_trigger_period()?;
+
+            let event = self.parse_keyword_separated(Keyword::OR, Parser::parse_trigger_event)?;
+            let table_name = if self.parse_keyword(Keyword::ON) {
+                self.parse_object_name()?
+            } else {
+                return self.expected("keyword `ON`", self.peek_token());
+            };
+
+            let mut referencing = vec![];
+            if self.parse_keyword(Keyword::REFERENCING) {
+                while let Some(refer) = self.parse_trigger_referencing()? {
+                    referencing.push(refer);
+                }
+            }
+
+            let for_each = if self.parse_keywords(&[Keyword::FOR]) {
+                let _ = self.parse_keyword(Keyword::EACH);
+                match self.parse_one_of_keywords(&[Keyword::ROW, Keyword::STATEMENT]) {
+                    Some(Keyword::ROW) => Some(TriggerObject::Row),
+                    Some(Keyword::STATEMENT) => Some(TriggerObject::Statement),
+                    _ => {
+                        return self.expected("an `ROW` OR `STATEMENT`", self.peek_token());
+                    }
+                }
+            } else {
+                None
+            };
+
+            let condition = if self.parse_keyword(Keyword::WHEN) {
+                self.expect_token(&Token::LParen)?;
+                let mut lparen_count = 1;
+                let mut condition_str = String::new();
+                loop {
+                    if lparen_count == 0 {
+                        break;
+                    }
+                    if let Some(next_token) = self.next_token_no_skip() {
+                        match &next_token.token {
+                            Token::LParen => lparen_count += 1,
+                            Token::RParen => lparen_count -= 1,
+                            Token::EOF => {
+                                return self.expected(" `)` ", TokenWithLocation::wrap(Token::EOF));
+                            }
+                            _ => {}
+                        }
+                        if lparen_count == 0 {
+                            break;
+                        }
+                        condition_str.push_str(next_token.token.to_string().as_str());
+                    }
+                }
+                Some(condition_str)
+            } else {
+                None
+            };
```

Comment:
I think this code would become much simpler when `condition` is an `Expr`

Something like `let condition = self.parse_expr()`